drivers/sensor/nxp/fxls8974/fxls8974.c

```diff
@@ -491,7 +491,7 @@ static int fxls8974_init(const struct device *dev)
 			return -EIO;
 		}
 
-		if (data->whoami == WHOAMI_ID_FXLS8974) {
+		if (data->whoami == cfg->expected_whoami_value) {
```

Comment:
I'm a bit confused by this. `data->whoami` is only used in this function, there's no reason to keep it around, why not just create a local variable in the init function for it?

The normal pattern for drivers is to just check for the valid ones supported so

```c
if (whoami != WHOAMI_ID_FXLS8974 && whoami != SOME_OTHER_ID) {
  LOG_ERR("...");
  return -ENXIO;
}
```

Reply:
I have updated the PR, please let me know if you have other suggestions